beam.py

```diff
@@ -135,23 +159,43 @@ def authenticate(self, *args):
 
             self.send_message(*args[:-1], method="auth")
 
+            if self.quiet:
+                self.http_session = Session()
+
             self.read_chat(self.handle)
         else:
-            raise ConnectionError(future.exception())
+            self.logger.error("There was an issue connecting. Trying again")
+            self.authenticate(*args)
```

Comment:
Definitely agree - exponential backoff.
I happen to have that implemented in `v0.4` - feel free to move that into here. :smile:
https://github.com/CactusBot/CactusBot/blob/refactor/cactusbot/services/websocket.py#L33